tests/test_dataset_signatures.py

```diff
@@ -0,0 +1,53 @@
+from inspect import signature
+
+import pandas as pd
+import pytest
+from pandas.testing import assert_frame_equal
+
+import reciprocalspaceship as rs
+
+
+def test_reset_index_dataseries():
+    """
+    Minimal example from GH#223
+    """
+    result = rs.DataSeries(range(10)).reset_index()
+    expected = pd.Series(range(10)).reset_index()
+    expected = rs.DataSet(expected)
+    assert_frame_equal(result, expected)
+
+
+def test_reset_index_signature(dataset_hkl):
+    """
+    Test call signature of rs.DataSet.reset_index() matches call signature of
+    pd.DataFrame.reset_index() using default parameters
+    """
+    df = pd.DataFrame(dataset_hkl)
+    sig = signature(pd.DataFrame.reset_index)
+    bsig = sig.bind(df)
+    bsig.apply_defaults()
+
+    expected = df.reset_index(*bsig.args[1:], **bsig.kwargs)
```

Comment:
You test the results, but is there any way to directly test the equivalency of the signatures?

Reply:
I could go through the `bsig.parameters.keys()` and make sure they match in the same order. I thought of this test as "good enough" because it would flag any changes to argument names. I like the idea that in the future we could add new arguments/default values if we need to, which is why I didn't quite want to directly assess signatures.

Reply:
okay. sounds fine for now.